WordPress/src/main/java/org/wordpress/android/ui/posts/MediaUploadCompletionProcessorPatterns.java

```diff
@@ -0,0 +1,214 @@
+
+    /**
+     * Helper class with methods to generate patterns for extracting and splicing text from raw block contents
+     */
+    static class Helpers {
+        /**
+         * Creates a pattern to extract image block contents and splice the header with a remote id
+         *
+         * @param localId the local image id to be matched
+         * @return A {@link Pattern} with the following capture groups:
+         *
+         * <ol>
+         *     <li>Block header before id</li>
+         *     <li>The localId</li>
+         *     <li>Block header after id</li>
+         *     <li>Block contents</li>
+         *     <li>Block closing comment and any following characters</li>
+         * </ol>
+         */
+        public static Pattern getImageBlockPattern(String localId) {
+            return Pattern.compile(String.format(PATTERN_TEMPLATE_IMAGE, localId), Pattern.DOTALL);
+        }
+
+        /**
+         * Creates a pattern to extract video block contents and splice the header with a remote id
+         *
+         * @param localId the local image id to be matched
+         * @return A {@link Pattern} with the following capture groups:
+         *
+         * <ol>
+         *     <li>Block header before id</li>
+         *     <li>The localId</li>
+         *     <li>Block header after id</li>
+         *     <li>Block contents</li>
+         *     <li>Block closing comment and any following characters</li>
+         * </ol>
+         */
+        public static Pattern getVideoBlockPattern(String localId) {
+            return Pattern.compile(String.format(PATTERN_TEMPLATE_VIDEO, localId), Pattern.DOTALL);
+        }
+
+        /**
+         * Creates a pattern to extract media-text block contents and splice the header with a remote id
+         *
+         * @param localId the local image id to be matched
+         * @return A {@link Pattern} with the following capture groups:
+         *
+         * <ol>
+         *     <li>Block header before id</li>
+         *     <li>The localId</li>
+         *     <li>Block header after id</li>
+         *     <li>Block contents</li>
+         *     <li>Block closing comment and any following characters</li>
+         * </ol>
+         */
+        public static Pattern getMediaTextBlockPattern(String localId) {
+            return Pattern.compile(String.format(PATTERN_TEMPLATE_MEDIA_TEXT, localId), Pattern.DOTALL);
+        }
+
+        /**
+         * Creates a pattern to extract gallery block contents and splice the header with a remote id
+         *
+         * @param localId the local image id to be matched
+         * @return A {@link Pattern} with the following capture groups:
+         *
+         * <ol>
+         *     <li>Block header before id</li>
+         *     <li>localId</li>
+         *     <li>Block header after id</li>
+         *     <li>Block contents</li>
+         *     <li>Block closing comment and any following characters</li>
+         * </ol>
+         */
```

Comment:
independent of the previous comment, there's some duplication here, both on comments and function implementations. Can we reduce this?

first thing that pops into my head: Instead of different getXXXBlockPattern functions maybe we can do this:

```

 public static Pattern getPattern(String localId, MediaBlockType type) {
   return Pattern.compile(String.format(getTemplate(type), localId), Pattern.DOTALL);
 }

String getTemplate(MediaBlockType type) {
  // switch type...
}

```

Reply:
> there's some duplication here, both on comments and function implementations. Can we reduce this?

Absolutely! I've refactored this to DRY up the pattern usage, moving most of the logic to the parent (abstract) `BlockProcessor` class.